cmd/scheduler.go

```diff
@@ -4,36 +4,50 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"os"
 
 	"github.com/frain-dev/convoy"
 	"github.com/frain-dev/convoy/config"
 	"github.com/frain-dev/convoy/internal/pkg/metrics"
+	"github.com/frain-dev/convoy/pkg/log"
 	redisqueue "github.com/frain-dev/convoy/queue/redis"
+	"github.com/frain-dev/convoy/util"
 	"github.com/frain-dev/convoy/worker"
 	"github.com/go-chi/chi/v5"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
-	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
 func addSchedulerCommand(a *app) *cobra.Command {
 	var exportCronSpec string
 	var port uint32
+	var logLevel string
 	cmd := &cobra.Command{
 		Use:   "scheduler",
 		Short: "scheduler runs periodic tasks",
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error {
 			cfg, err := config.Get()
 			if err != nil {
-				log.Fatalf("Error getting config: %v", err)
+				a.logger.Errorf("Failed to retrieve config: %v", err)
+				return err
 			}
-			if cfg.Queue.Type != config.RedisQueueProvider {
-				log.WithError(err).Fatalf("Queue type error: Command is available for redis queue only.")
+
+			if util.IsStringEmpty(logLevel) {
+				logLevel = cfg.Logger.Level
+			}
+
+			lo := log.NewLogger(os.Stdout, "scheduler")
```

Comment:
What's the difference between `a.logger` and this one?

Reply:
`a.logger` is used only by the `main` package. It is used only during app startup. This one is passed down and used by the `scheduler`.

Reply:
Hmm, ok, but when starting up the process aren't we just running a single command? Is there a need to configure two different loggers?